### ISSUE TITLE
AssertionFailedException does not cross AppDomain boundaries

### ISSUE THREAD

**Issue by ECrank** [dennisdoomen/fluentassertions#214]
I think this Exception class should be serializable so that you can run tests in a separate AppDomain and a failure exception will be reported in the unit test AppDomain. I am using the following pattern to perform tests that rely on uninitialized static state:
```C#
AppDomain domain = AppDomain.CreateDomain("TestSingletonTrue", AppDomain.CurrentDomain.Evidence, AppDomain.CurrentDomain.SetupInformation);
try
{
    domain.DoCallBack(() =>
    {
        // Do Test
    });
}
finally
{
    AppDomain.Unload(domain);
}
```
On failure, I get the following message: 
>System.Runtime.Serialization.SerializationExceptionType 'FluentAssertions.Execution.AssertionFailedException' in assembly 'FluentAssertions, Version=2.0.1.0, Culture=neutral, PublicKeyToken=33f2691a05b67b6a' is not marked as serializable.

**Comment by vossad01** [comment]
I see no reason that `AssertionFailedException` cannot be made serializable.

Do you think you could prepare a Pull Request for that change?

In case you are not familiar with the behavior, I should mention that the type of exception thrown will vary depending on the test framework (NUnit, MSTest, etc.) that was detected or configured.  So long as FA is using the [`FallbackTestFramework`](https://github.com/dennisdoomen/fluentassertions/blob/5b3eaa336e2fb347afe2cfe29f9b3c5a1153b8f6/Shared/Execution/FallbackTestFramework.cs) it will use `AssertionFailedException`.

**Comment by konstmsu** [comment]
Microsoft guidelines suggest exceptions should be  serializable so this is
definitely the right thing to do
http://msdn.microsoft.com/en-us/library/vstudio/ms229064%28v=vs.100%29.aspx

**Comment by dennisdoomen** [comment]
Fixed by PR #228.

**Comment by florianwittmann** [comment]
Actually PR #228 is missing the serialization constructor as described here: http://stackoverflow.com/questions/94488/what-is-the-correct-way-to-make-a-custom-net-exception-serializable
Therefore currently it still can not be used cross-appdomain as deserialization is failing.
